From fa6268e71b717f0c5defe2a1f65c710d09f26bc3 Mon Sep 17 00:00:00 2001
From: Eric Bentley <eric.bentley@lairdtech.com>
Date: Tue, 14 Jan 2014 15:11:16 -0500
Subject: [PATCH] Save the firmware string for later retrieval

The original catalyst for the bug was CCX status needed to be determined
by the cli on the WB40.  The necessitated getting the firmware string
from the driver.  This change gives the same capability to the 45
driver.

The firmware version is displayed in sdc_cli version.

The driver version is set to 3.5.0.0 to sync with latest SDK

BZ 4670
---
 drivers/net/wireless/ath/ath6kl/wmi.c   | 10 ++++++++++
 drivers/net/wireless/laird_fips/laird.h |  2 +-
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath6kl/wmi.c b/drivers/net/wireless/ath/ath6kl/wmi.c
index 9427836..6f64307 100644
--- a/drivers/net/wireless/ath/ath6kl/wmi.c
+++ b/drivers/net/wireless/ath/ath6kl/wmi.c
@@ -84,6 +84,7 @@ enum { // keep this list in sync with the one from ath_access.c from the SDK
 	GETTXPOWER,
 	GETAPNAME,
 	GETAPIP,
+	GETFWSTR,
 } ATHEROS_CMD_GET_VALUES;
 
 #define AIRONET_CCX_IE   0x85
@@ -4463,6 +4464,7 @@ static int ath6kl_genl_get_value (struct sk_buff *skb_2, struct genl_info *info)
 	struct ath6kl *ar;  
 	struct wmi *wmi = gwmi;	
 	struct nlattr *na;
+	char fwStr[80];
 
 	if ( gwmi == NULL )
 		return 0;
@@ -4505,6 +4507,14 @@ static int ath6kl_genl_get_value (struct sk_buff *skb_2, struct genl_info *info)
 						ar->laird.AP_IP[2] << 8 |
 						ar->laird.AP_IP[3]);
 				break;
+			case GETFWSTR:
+				snprintf(fwStr, 80, "%s fw %s api %d%s",
+						ar->hw.name,
+						ar->wiphy->fw_version,
+						ar->fw_api,
+						test_bit(TESTMODE, &ar->flag) ? " testmode" : "");
+				rc = nla_put_string(skb, ATHEROS_ATTR_MSG, fwStr);
+				break;
 		}
 	}
 	if (rc != 0)
diff --git a/drivers/net/wireless/laird_fips/laird.h b/drivers/net/wireless/laird_fips/laird.h
index 7153f5b..dc3ec62 100644
--- a/drivers/net/wireless/laird_fips/laird.h
+++ b/drivers/net/wireless/laird_fips/laird.h
@@ -9,7 +9,7 @@
 /* Laird version is 32bit value.  Parsed in the form w.x.y.z.
  * increment y.z as needed for each change
  */
-#define LAIRD_DRV_VERSION 0x03040006
+#define LAIRD_DRV_VERSION 0x03050000
 
 #ifdef LAIRD_FIPS
 #include <linux/etherdevice.h>
-- 
1.8.1.2

